### ISSUE TITLE
After updating permission, resource does not exist, locked out of console.

### ISSUE THREAD

**Issue by GimpMaster** [hasura/graphql-engine#8944]
### Version Information

Server Version: 2.10.0
CLI Version (for CLI related issue):

### Environment

OSS: Hosted on an Ubuntu 20.04 LTS server. Docker container (compose script with both postgres and hasura)

### What is the current behaviour?

Whenever I go to update permissions in the database (i.e. add a new permission role) the hasura console locks me out. The only way to get back into hasura console is to reboot the machine. Restarting the docker container doesn't help.

Looking at the errors on the network tab I get the following 404 errors:

404 schema/public/permission/v1/metadata

```
code: "not-found"
error: "resource does not exist"
path: "$"
```

the docker logs show the following:

```
{"detail":{"http_info":{"content_encoding":null,"http_version":"HTTP/1.1","ip":"X.X.X.X","method":"POST","status":404,"url":"/console/data/PM%20Tool/schema/public/permission/v1/metadata"},"operation":{"error":{"code":"not-found","error":"resource does not exist","path":"$"},"raw_query":"{\"type\":\"get_inconsistent_metadata\",\"args\":{}}","request_id":"42f8835e-731e-497c-bd1d-6ecf8bd72929","request_mode":"error","response_size":65},"request_id":"42f8835e-731e-497c-bd1d-6ecf8bd72929"},"level":"error","timestamp":"2022-09-08T04:37:20.866+0000","type":"http-log"}
```

Again hasura has been operating great for the past month. Its just whenever I try to make a permission change does it act up.

Any thouhgts?

### What is the expected behaviour?

That I can change permissions without having to reboot my server.

### Keywords

not-found, resouce does not exist.

**Comment by GimpMaster** [comment]
Additional notes: I have a local hasura docker container pointing to the remote postgres. When this issue happens the local one still continues working.....UNTIL i try to adjust the permissions on the local console. Then the local starts saying resource: not-found. 

The local one is running in a WSL2 instances. I have to do a full wsl --shutdown. Then I can bring back up hasura and have it start working again.

**Comment by adas98012** [comment]
From the docker log it seems you have inconsistent metadata. 
You may be able to resolve it and then proceed with your workflow.
Here is the Hasura Document to spot inconsistent metedata (and then resolve it):

https://hasura.io/docs/latest/api-reference/metadata-api/manage-metadata/#metadata-get-inconsistent-metadata

**Comment by adas98012** [comment]
And here is the same info for Hasura CLI:

https://hasura.io/docs/latest/hasura-cli/commands/hasura_metadata_inconsistency_list/

**Comment by GimpMaster** [comment]
Thank you.

Do you know if you can run the hasura cli from inside a docker container?

Or is it a tool that lives on the host?

**Comment by GimpMaster** [comment]
@adas98012  - I was able to get hasura cli installed and with the config.yaml I was able to connect to my remote docker endpoint url. The hasura metadata commands run as expected. No inconsistency listing. 

So I logged in again. I copied permissions from one role to another (new role). It copied it successfully. Then when clicking on any link in the console it instantly logged me out. I get the resource not found.

One thing I found interesting is that this is the URL it is calling

/console/data/PM Tool/schema/public/permission/console/logi/v1/metadata

is the URL really `logi/v1` and not `login/v1` ?

Does it not like the space in the PM Tool hasura instance name?
